src/core/Tags.java

```diff
@@ -637,4 +637,19 @@ public static boolean looksLikeInteger(final String value) {
     return true;
   }
 
+  /**
+   * Returns true if the given string can fit into a float.
+   * @param value The String holding the float value.
+   * @return true if the value can fit into a float, false otherwise.
+   * @throws NumberFormatException if the value is not numeric.
+   * @since 2.0.2
+   */
+  public static boolean fitsInFloat(final String value) {
+    final float f = Float.parseFloat(value);
+    final String converted = Float.toString(f);
+
+    // this will be false if there was a loss of precision.
+    return value.equals(converted);
```

Comment:
So there are two ways that doubles represent more values - the exponent range is larger, and it has more mantissa bits.  The exponent range is easy enough, and @manolama 's suggestion of casting and comparing to infinity will catch that (not sure what the issue is with negative floats -  a value less than Float.MIN_VALUE should be converted to Float.NEGATIVE_INFINITY; if this really is an issue, then you can look at Double.doubleToLongBits and extract the exponent yourself).  The only way to be certain you aren't losing any precision is if the extra mantissa bits are all 0, but in practice representing decimal values that is not ever going to be the case, because most decimal fractions are infinitely long repeating binary fractions.  Instead, what you want is some acceptable level of precision loss, and it probably is an absolute value (ie, if the difference is 1e-10 you probably don't care, while if it is 1e10 you probably do, even if the number of bits in the mantissa being lost is the same.  So my suggestion for the test would be casting to float and check if the absolute value difference is greater than .0078125 (2^-7), but this really should be configurable since it depends on what data I am storing (and actually should be per-metric, but that is probably unacceptable overhead).